File: src/types/receipt.ts
export interface Receipt {
  id: string
  userId: string
  vendor: string
  date: string
  total: number
  imageUrl?: string
  createdAt: string
  updatedAt: string
  items: ReceiptItem[]
}

export interface ReceiptItem {
  id: string
  receiptId: string
  productId: string
  quantity: number
  unitPrice: number
  totalPrice: number
  expenseTag?: string | null
  trackQuantity?: number | null
  quantityUnit?: string | null
  createdAt: string
  product: {
    id: string
    name: string
    category: string
    brand?: {
      name: string
    }
  }
}

File: python-backend/app/models/receipt.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class ReceiptItem(BaseModel):
    name: str
    quantity: float
    unit_price: float
    total_price: float
    category: Optional[str] = "Other"

class ParsedReceiptData(BaseModel):
    vendor: str
    date: str
    total: float
    items: List[ReceiptItem]
    raw_text: Optional[str] = None
    confidence: Optional[float] = None

class ReceiptProcessRequest(BaseModel):
    image_base64: str
    
class ReceiptProcessResponse(BaseModel):
    success: bool
    data: Optional[ParsedReceiptData] = None
    error: Optional[str] = None